airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialLoadHandler.java

```diff
@@ -0,0 +1,161 @@
+import io.airbyte.cdk.integrations.source.relationaldb.TableInfo;
+import io.airbyte.cdk.integrations.source.relationaldb.models.OrderedColumnLoadStatus;
+import io.airbyte.commons.stream.AirbyteStreamUtils;
+import io.airbyte.commons.util.AutoCloseableIterator;
+import io.airbyte.commons.util.AutoCloseableIterators;
+import io.airbyte.integrations.source.mssql.MssqlSourceOperations;
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.CommonField;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteMessage.Type;
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
+import io.airbyte.protocol.models.v0.AirbyteStream;
+import io.airbyte.protocol.models.v0.CatalogHelpers;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.v0.SyncMode;
+import java.sql.JDBCType;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MssqlInitialLoadHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MssqlInitialLoadHandler.class);
+  private static final long RECORD_LOGGING_SAMPLE_RATE = 1_000_000;
+  private final JsonNode config;
+  private final JdbcDatabase database;
+  private final MssqlSourceOperations sourceOperations;
+  private final String quoteString;
+  private final MssqlInitialLoadStateManager initialLoadStateManager;
+  private static final long DEFAULT_CHUNK_SIZE = 1_000_000;
+  private final Function<AirbyteStreamNameNamespacePair, JsonNode> streamStateForIncrementalRunSupplier;
+
+  public MssqlInitialLoadHandler(
+      final JsonNode config,
+      final JdbcDatabase database,
+      final MssqlSourceOperations sourceOperations,
+      final String quoteString,
+      final MssqlInitialLoadStateManager initialLoadStateManager,
+      final Function<AirbyteStreamNameNamespacePair, JsonNode> streamStateForIncrementalRunSupplier) {
+    this.config = config;
+    this.database = database;
+    this.sourceOperations = sourceOperations;
+    this.quoteString = quoteString;
+    this.initialLoadStateManager = initialLoadStateManager;
+    this.streamStateForIncrementalRunSupplier = streamStateForIncrementalRunSupplier;
+  }
+  public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
+      final ConfiguredAirbyteCatalog catalog,
+      final Map<String, TableInfo<CommonField<JDBCType>>> tableNameToTable,
+      final Instant emittedAt) {
+    final List<AutoCloseableIterator<AirbyteMessage>> iteratorList = new ArrayList<>();
```

Comment:
This is beyond the scope of this change but does spamming the `final` keyword really improve code quality? I wonder especially in cases like this one when we're assigning a value which is in reality obviously mutable, even if the enclosing container reference doesn't change.

Reply:
The narrowest response to why add `final` is that something will complain if you don't mark it as final and try to build.

It improves performance I think

Reply:
I tried removing `final` in a bunch of places and nothing complained. Perhaps that's because I ripped out PMD a few months ago (nobody noticed, I think).

I don't buy the performance argument. Very little of our code is in the hotpath. Not to mention that if PMD can figure out which variables are `final` then I trust that javac can tool.

IMHO lack of readability is a fantastic source of bugs and our code isn't great in this regard. Let's all improve it a little bit at a time.